test/distributed/org/apache/cassandra/distributed/test/JMXSslConfigDistributedTest.java

```diff
@@ -0,0 +1,205 @@
+                                                                            .put("enabled", true)
+                                                                            .put("require_client_auth", true)
+                                                                            .put("accepted_protocols", Arrays.asList("TLSv1.2", "TLSv1.3", "TLSv1.1"))
+                                                                            .build();
+
+            try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+                c.with(Feature.JMX);
+                c.set("jmx_encryption_options", encryptionOptionsMap);
+            }).start())
+            {
+                Map<String, Object> jmxEnv = new HashMap<>();
+                configureClientSocketFactory(jmxEnv, encryptionOptionsMap);
+                // Invoke the same code vs duplicating any code from the JMXGetterCheckTest
+                JMXGetterCheckTest.testAllValidGetters(cluster, jmxEnv);
+            }
+        }
+    }
+
+    @Test
+    public void testSystemSettings() throws Throwable
+    {
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL.setBoolean(true);
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_NEED_CLIENT_AUTH.setBoolean(false);
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_PROTOCOLS.setString("TLSv1.2,TLSv1.3,TLSv1.1");
+        COM_SUN_MANAGEMENT_JMXREMOTE_SSL_ENABLED_CIPHER_SUITES.reset();
+        try(WithProperties withProperties = new WithProperties().with("javax.net.ssl.trustStore", (String)validKeystore.get("truststore"),
+                                                                      "javax.net.ssl.trustStorePassword", (String)validKeystore.get("truststore_password"),
+                                                                      "javax.net.ssl.keyStore", (String)validKeystore.get("keystore"),
+                                                                      "javax.net.ssl.keyStorePassword", (String)validKeystore.get("keystore_password"))
+        )
+        {
+            try (Cluster cluster = builder().withNodes(1).withConfig(c -> {
+                c.with(Feature.JMX);
+            }).start())
+            {
+                Map<String, Object> jmxEnv = new HashMap<>();
+                SslRMIClientSocketFactory clientFactory = new SslRMIClientSocketFactory();
+                jmxEnv.put(RMIConnectorServer.RMI_CLIENT_SOCKET_FACTORY_ATTRIBUTE, clientFactory);
+                jmxEnv.put("com.sun.jndi.rmi.factory.socket", clientFactory);
+                // Invoke the same code vs duplicating any code from the JMXGetterCheckTest
+                JMXGetterCheckTest.testAllValidGetters(cluster, jmxEnv);
+            }
+        }
+    }
+
+    @Test
+    public void testInvalidKeystorePath() throws Throwable
+    {
+        ImmutableMap<String, Object> encryptionOptionsMap = ImmutableMap.<String, Object>builder()
```

Comment:
just put `encryptionOptionsMap` into c.set(...), just inline it all and everywhere

Reply:
oh I see. Initially I had some trouble setting this so this was left unattended. Now its fixed.

Reply:
@maulin-vasavada you haven't inlined this, `encryptionOptionsMap` are still outside of try catch. Just make it same as you did in `testDisabledEncryptionOptions`. `testClientAuth` and `testDefaultEncryptionOptions` are not reworked either.

Reply:
Fixed it.